src/main/java/com/hotels/beeju/extensions/HiveServer2JUnitExtension.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (C) 2015-2019 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.beeju.extensions;
+
+import java.util.Map;
+
+import org.apache.hive.jdbc.HiveDriver;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import com.hotels.beeju.core.HiveServer2Core;
+
+public class HiveServer2JUnitExtension extends BeejuJUnitExtension {
+
+  private final HiveServer2Core hiveServer2Core;
+
+  /**
+   * Create a HiveServer2 service with a pre-created database "test_database".
+   */
+  HiveServer2JUnitExtension() {
+    this("test_database");
+  }
+
+  /**
+   * Create a HiveServer2 service with a pre-created database using the provided name.
+   *
+   * @param databaseName Database name.
+   */
+  HiveServer2JUnitExtension(String databaseName) {
+    this(databaseName, null);
+  }
+
+  /**
+   * Create a HiveServer2 service with a pre-created database using the provided name and configuration.
+   *
+   * @param databaseName Database name.
+   * @param configuration Hive configuration properties.
+   */
+  HiveServer2JUnitExtension(String databaseName, Map<String, String> configuration) {
+    super(databaseName, configuration);
+    hiveServer2Core = new HiveServer2Core(core);
+  }
+
+  @Override
+  protected void init() throws Exception {
```

Comment:
OK, so we override `init()` here, is there any reason this can't go into the `beforeEach()` here instead? I think it would be clearer exactly what order things happen in this way.

Reply:
I think previously the temporary folder stuff was created in here but seeing as we've moved that out, I guess there isn't really any reason to have the method anymore, so I'll move the methods here into beforeEach(), because I don't think the order ends up changing